src/west/manifest.py

```diff
@@ -1774,28 +1774,28 @@ def _load_imap(self, imp: Dict, src: str) -> _import_map:
         # Work on a copy in case the caller needs the full value.
         copy = dict(imp)
         ret = _import_map(copy.pop('file', _WEST_YML),
-                          copy.pop('name-whitelist', []),
-                          copy.pop('path-whitelist', []),
-                          copy.pop('name-blacklist', []),
-                          copy.pop('path-blacklist', []),
+                          copy.pop('name-allowlist', []),
+                          copy.pop('path-allowlist', []),
+                          copy.pop('name-denylist', []),
+                          copy.pop('path-denylist', []),
```

Comment:
I'm glad to see you getting ahead of this, but we can't implement it this way. We have to maintain backwards compatibility or NCS will break. Users (including NCS) have manifests in their history with the white/black terms and I think it's not OK to break those.

Can you please rework this so that the old values are still used as a fallback, and leave a comment explaining why?

We can then update the west documentation to only document the new way of doing things.

Reply:
I'd say west is a `0.x` library so breaking changes are accepted, but you are right: this can be handled in a non-breaking fashion, by keeping the support for the deprecated terms and printing a warning.

I'll look into it.

Reply:
> I'd say west is a 0.x library so breaking changes are accepted,

The version number not being 1.0 is about the APIs being unstable, as is stated in the docs. The command line interface and manifest parsing needs to stay compatible with what people are already using unless it is impossible to do otherwise. Remember that west.yml is part of git history and people can do bisections across large commit ranges. It's not acceptable to introduce a break like this.

> but you are right: this can be handled in a non-breaking fashion, by keeping the support for the deprecated terms and printing a warning.

A warning is also not acceptable, for the same reason. Just silently falling back on the old names.

Thanks!

Reply:
I don't get why changing a file in a repo's `west.yaml` breaks git bisect operations for this repository.

However I believe you that it does and therefore adding support for both variants is the only way to do it right.

Reply:
> I don't get why changing a file in a repo's `west.yaml` breaks git bisect operations for this repository.
> 

Manifest repo, as example `nrf-repo`.

> HEAD: bad (manifest: `allow-list`)
> ...
> HEAD\~25
> ...
> HEAD\~50: unknown (manifest: `white-list`)
> ...
> HEAD\~75
> ...
> HEAD\~100: good (manifest: `white-list`)

Starting a bisect:
```
git bisect start
git bisect bad
git bisect good HEAD~100
Bisecting: 49 revisions left to test after this (roughly 6 steps)
[HEAD~75] <text>
west update (this fails, cause latest `west` doesn't support `white-list` --> downgrade west)
pip install west==0.7.3
<test>
git bisect good
Bisecting: 24 revisions left to test after this (roughly 5 steps)
[HEAD~25] <text>
west update (this fails, cause old `west` doesn't support `allow-list` --> upgrade west)
pip install west==0.x.y
west update
<test>
git bisect bad
... --> maybe west needs to be downgraded this time, maybe not ?
```

you start to see how annoying this is to users, cause they won't know when a SHA that requires a different `west` version is needed, until `west update` fails.

Reply:
> A warning is also not acceptable, for the same reason. Just silently falling back on the old names is what I am asking for.

Agree.

Reply:
I think `include`/`exclude` would be more correct.
We are in fact including a project list/path, or excluding it from import.

```suggestion
                          copy.pop('name-includelist', []),
                          copy.pop('path-includelist', []),
                          copy.pop('name-excludelist', []),
                          copy.pop('path-exludelist', []),
```

Reply:
Honestly if we're going to bikeshed this I would rather use 'allow' and 'block' for consistency with what other major projects seem to be doing: https://www.theregister.com/2019/09/03/chromium_microsoft_offensive/

I don't think there's any correctness issue, though.

Reply:
> Honestly if we're going to bikeshed this I would rather use 'allow' and 'block' for consistency with what other major projects seem to be doing

Consistency is king. Accepted.

Wonder why `include`/`exclude` is not used, maybe `exclude` is too offensive for all those poor software developers (like myself) that sucked when playing football/soccer as child so we were always excluded from playing :thinking:

Reply:
If used `includelist`/`excludelist` and kept support for `whitelist`/`blacklist` in the latest push.